### DISCUSSION TITLE
30 seconds faster CI feedback on your pull requests

### DISCUSSION THREAD

**Discussion by peterbe** [mdn/mdn-community#490] [Content]
**tl;dr; PRs finish roughly ~30 seconds faster as of yesterday.**

When you create a PR, it's highly productive to be able to find out if it failed/passed as soon as possible. Otherwise, the mind wanders and you might be too disengaged or bored to finish what wasn't working. And if it did pass, you'll want to see the Preview URLs sooner from the PR Review Companion. 

Here's the original issue: https://github.com/mdn/content/issues/8014
Here's the solution: https://github.com/mdn/content/pull/8066

Basically, we use `actions/cache` to cache the whole `node_modules` between runs. If `yarn.lock` hasn't changed, `node_modules/` shouldn't change. Being able to skip the `yarn install` step can save you roughly 30 seconds because retrieving from cache is 1-4seconds. 

[The revelation](https://github.community/t/how-come-the-actions-cache-is-failing-so-much-often/196279/3) was that a PR can't re-use a cached thing unless it was previously stored on the `main` branch (or its own branch when 1 PR is multiple commits). So now, after your PR is merged, the CI tests run one more time, in the background once it's been merged in `main`.

**Comment by pepelsbey** [comment]
Closing as answered/settled.
